train.py

```diff
@@ -1,203 +1,45 @@
-    PUPPI_pt = normFac * np.sum(Xr_valid[0][:,:,4:6], axis=1)
-    predict_test = predict_test *normFac
-    Yr_valid = normFac * Yr_valid
-    #Xr_valid = normFac * Xr_valid
-
-    #test_events = Xr_valid[0].shape[0]
-
-    MakePlots(Yr_valid, predict_test, PUPPI_pt, path_out = path_out)
-    
-    Yr_valid = convertXY2PtPhi(Yr_valid)
-    predict_test = convertXY2PtPhi(predict_test)
-    PUPPI_pt = convertXY2PtPhi(PUPPI_pt)
-
-    MET_rel_error_opaque(predict_test[:,0], PUPPI_pt[:,0], Yr_valid[:,0], name=''+path_out+'rel_error_opaque.png')
-    MET_binned_predict_mean_opaque(predict_test[:,0], PUPPI_pt[:,0], Yr_valid[:,0], 20, 0, 500, 0, '.', name=''+path_out+'PrVSGen.png')
-    extract_result(predict_test, Yr_valid, path_out, 'TTbar', 'ML')
-    extract_result(PUPPI_pt, Yr_valid, path_out, 'TTbar', 'PU')
-    fi = open("{}time.txt".format(path_out), 'w')
-
-    fi.write("Working Time (s) : {}".format(end_time - start_time))
-    fi.write("Working Time (m) : {}".format((end_time - start_time)/60.))
-
-    fi.close()
-
-
-# Configuration
 
 if __name__ == "__main__":
-
     time_path = time.strftime('%Y-%m-%d', time.localtime(time.time()))
     path = "./result/"+time_path+"_PUPPICandidates/"
-
+    
+    # 4 arguments
+    # train from h5 or root?
+    # root file location
+    # output path
+    # mode 0 or 1
     parser = argparse.ArgumentParser()
-    parser.add_argument('--input', action='store', type=str, required=True, help='designate input file path')
-    parser.add_argument('--output', action='store', type=str, default='{}'.format(path), help='designate output file path')
-    parser.add_argument('--mode', action='store', type=int, required=True, help='0 for L1MET, 1 for DeepMET')
-        
+    parser.add_argument('--dataSetType', action='store', type=str, required=True, help='designate input file path')
+    parser.add_argument('--input', action='store', type=str, required=False, help='designate input file path')
+    parser.add_argument('--output', action='store', type=str, required=True, help='designate output file path')
+    parser.add_argument('--mode', action='store',   type=int, required=True, help='0 for L1MET, 1 for DeepMET')
     args = parser.parse_args()
+
+dataSetType = args.dataSetType
+if dataSetType == 'h5':
+    inputPath = args.input
+    # convert root files to h5 and store in same location
+    i =0
+    for file in os.listdir(inputPath):
+        if '.root' in file:
+            os.system(f'python convertNanoToHDF5_L1triggerToDeepMET.py -i {inputPath}/{file} -o {inputPath}/set{i}.h5')
+            i += 1
+    # place h5 file names into a .txt file
+    writeFile= open(f'{inputPath}/h5files.txt',"w+")
+    for file in os.listdir(inputPath):
+        if '.h5' in file:
+            writeFile.write(f'{inputPath}/{file}\n')
+    writeFile.close()
+    # this file is read in main()
+    h5files = f'{inputPath}/h5files.txt'
+    h5filesList = f'{inputPath}/h5files.txt'
+            
+    from train_fromh5 import main
```

Comment:
it would be better to name the `main` function something more meaningful like `train_fromh5`